pkg/api/v1/schema.go

```diff
@@ -35,8 +35,20 @@ type DomainSpec struct {
 }
 
 type Memory struct {
-	Value uint   `json:"value"`
-	Unit  string `json:"unit"`
+	// Allocated field defines the amount of memory a VM gets upon starting in kiloBytes
+	// The field type means the total maximum is 4 TiB of memory
+	Allocated uint `json:"allocated"`
+
+	// Required fields defines the amount of memory a VM needs to operate properly.
+	// This field can be set to the same value as `Allocated` to effectively
+	// disable any memory overcommit features. The value is provided in Bytes.
+	Required uint `json:"required"`
```

Comment:
We don't have an overcommit feature yet. Since this value doesn't do anything yet, maybe we shouldn't have it until memory overcommit is thought through more.

Reply:
We do have overcommit actually, because currently we do not limit at all :) We do not support ballooning or KSM atm, but it does not mean all VMs will use their full memory (they don't mostly). And so we can easily allow simple aggregation (telco term, basically overselling the capacity and hoping people will take turns in using it.) without anything special.